src/events/upgrade.py

```diff
@@ -33,85 +38,108 @@ class KafkaDependencyModel(BaseModel):
     kafka_service: DependencyModel
 
 
-class KafkaUpgrade(DataUpgrade):
+class UpgradeManager(DataUpgrade):
     """Implementation of :class:`DataUpgrade` overrides for in-place upgrades."""
 
     def __init__(self, charm: "KafkaK8sCharm", **kwargs):
         super().__init__(charm, **kwargs)
         self.charm = charm
+        self.substrate = self.charm.state.substrate
 
-    @property
-    def current_version(self) -> str:
-        """Get current Kafka version."""
-        dependency_model: DependencyModel = getattr(self.dependency_model, "kafka_service")
-        return dependency_model.version
```

Comment:
**todo:** This property needs to be used. The upgrade is missing the `inter.broker.protocol.version` addition to `server.properties`. [See here](https://github.com/canonical/kafka-operator/pull/127/files#diff-24a3fe5c783755412cb9de83238b4dfd0699345bf50dcadc3ec23c6749361592R456-R465)